### PR TITLE
Use dockerized environment for running GH tests

### PR DESCRIPTION
**Pull request by jgadling**
This PR makes a few important changes:

- CI tests now run against the dockerized local-dev environment
- Docker image caching improvements that should improve image build/pull/push times
- `make local-database` is gone now. `make unit-test` now runs inside the docker environment
- Each successful build of the main branch will push a new image to the corpora-upload-dev ecr repo. I could use a little more info on which tags we want to use here.

#### Reviewers
**Functional:** 

**Readability:** 

---

## Changes
- add
- remove
- modify

### CODE REVIEW

**Review by Bento007 — COMMENTED** [review]
was a new ECR created for the test docker images?

Does every push-test upload a container to ECR?

File: .github/workflows/push-tests.yml [hunk 1]
```diff
@@ -53,40 +51,47 @@ jobs:
...
+        with:
+          aws-access-key-id: ${{ secrets.AWS_ACCESS_KEY }}
+          aws-secret-access-key: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
+          aws-region: us-west-2
+          role-to-assume: ${{ secrets.AWS_ROLE_TO_ASSUME }}
+          role-duration-seconds: 900
+      - name: Login to ECR
+        uses: docker/login-action@v1
+        with:
+          registry: ${{ secrets.ECR_REPO }}
```
[the remark this answers is not in the document]
Comment (jgadling) [reply .github/workflows/push-tests.yml:85]: This authenticates the GitHub Action to the ECR endpoint for (all repos in) the dev AWS account.

**Review by MDunitz — APPROVED** [review]
lgtm

**Review by Bento007 — APPROVED** [review]
LGTM pending a few changes we discussed.

**Pull request merged**